### PR TITLE
Add ability to change service types

### PR DESCRIPTION
**Pull request by jakalkhalili**

### CODE REVIEW

File: chart/jenkins-operator/templates/service.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (tomaszsek) [inline chart/jenkins-operator/templates/service.yaml:1]: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/schema/#github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2.JenkinsSpec please look at  `service` and `slaveService` 
The operator creates both services so you have to delete this file and map fields to Jenkins CR.

File: chart/jenkins-operator/values.yaml [hunk 2]
```diff
@@ -184,6 +184,16 @@ jenkins:
     # secretData creates new secret if secretRefName is empty and fills with data provided in secretData
     secretData: {}
 
+  services:
+    http:
+      port: 8080
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:189]: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/schema/#github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2.Service there are much more field to map

File: chart/jenkins-operator/values.yaml [hunk 3]
```diff
@@ -184,6 +184,16 @@ jenkins:
...
+      type: ClusterIP
+      nodePort: ""
+    slave:
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:192]: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/schema/#github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2.Service there are much more field to map

File: chart/jenkins-operator/values.yaml [hunk 4]
```diff
@@ -184,6 +184,16 @@ jenkins:
...
+      port: 5000
```
Comment (tomaszsek) [inline chart/jenkins-operator/values.yaml:193]:
```suggestion
      port: 50000
```

File: chart/jenkins-operator/templates/jenkins.yaml [hunk 5]
```diff
@@ -48,6 +48,18 @@ spec:
     recoveryOnce: {{ .Values.jenkins.backup.recoveryOnce }}
     {{- end }}
   {{- end }}
+  {{- with .Values.jenkins.services.http }}
```
Comment (snooyen) [inline chart/jenkins-operator/templates/jenkins.yaml:51]: @tomaszsek 
@jakalkhalili 

Would it not be better to just do the following? 
```
  {{- with .Values.jenkins.service }}
  service: {{- toYaml . | nindent 4 }}
  {{- end }}
  {{- with .Values.jenkins.slaveService }}
  slaveService: {{- toYaml . | nindent 4 }}
  {{- end }}
```

Comment (tomaszsek) [reply chart/jenkins-operator/templates/jenkins.yaml:51]: @snooyen Looks much better thanks.

**Pull request closed without merging**